lib/OpenQA/UserAgent.pm

```diff
@@ -65,6 +65,27 @@ sub new {
     return $self;
 }
 
+sub get {
+    my ($self, $url) = @_;
+
+    my $tx;
+
+    # Resolve the URL manually in case one of the routes doesn't work
+    use Socket;
+    my ($err, @addrs) = Socket::getaddrinfo($url->host);
+    for my $addr (@addrs) {
+        my ($err, $host) = Socket::getnameinfo($addr->{addr}, Socket::NI_NUMERICHOST);
+        $url->host($host);
+        $tx = $self->SUPER::get($url);
+        if (!$tx->error) {
+          last;
+        }
+        warn "failed to resolve $host: ".$tx->error->{message};
```

Comment:
Try to avoid warnings. If you want to signal an exception to the caller use a special return value such as `undef` or throw an exception. Nobody likes warnings they can't shut up. If you need this for debugging purposes it is customary to use an environment variable and declare a `DEBUG` constant at the top of the file.